### ISSUE TITLE
hfst-lookup crashes after > an hour on long compound word (North Sámi)

### ISSUE THREAD

**Issue by hfst-importer** [hfst/hfst#298]
```
$ time echo seksuálaláhkarihkkunlávdegottesesoŋŋaáigi | hfst-lookup -q tools/spellcheckers/fstbased/analyser-fstspeller-gt-norm.hfst 
Killed: 9

real	70m39.561s
user	69m42.073s
sys	0m43.796s
```

To repeat:

1) svn co https://gtsvn.uit.no/langtech/trunk/gtcore
2) ./autogen.sh && ./configure && make && sudo make install
3) svn -r123846 co https://gtsvn.uit.no/langtech/trunk/langs/sme
4) ./autogen.sh && ./configure --with-hfst --without-xfst --enable-spellers
5) make
6) echo seksuálaláhkarihkkunlávdegottesesoŋŋaáigi | hfst-lookup tools/spellcheckers/fstbased/analyser-fstspeller-gt-norm.hfst

The previous version did not crash, just took an enormous amount of time. The fst has some inserted flag diacritics to regulate compound behaviour.

Reported by: snomos

**Comment by eaxelson** [comment]
Bug #324 (as reported in Sourceforge bug tracker) has been fixed in commit 76e9069.

**Comment by eaxelson** [comment]
This got also fixed when bug #297 was fixed.
